.github/workflows/lint.yml

```
@@ -0,0 +1,35 @@
+name: Lint
+
+on: [push, pull_request]
+
+env:
+  # Configure pip to cache dependencies and do a user install
+  PIP_NO_CACHE_DIR: false
+  PIP_USER: 1
+  PYTHON_VERSION: 3.8
+
+  # Make sure package manager does not use virtualenv
+  POETRY_VIRTUALENVS_CREATE: false
+
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    name: pre-commit
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+
+      - name: Setup Python Environment
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.9"
+
+      - name: Setup Poetry
+        run: python -m pip install -U poetry
+
+      - name: Install Dependencies
+        run: poetry install
```

```suggestion
        run: poetry install --no-dev
```
No need for dev dependencies in CI. It takes extra time.

---

The CI does need dev deps. because `pre-commit`, `isort`, and `flake8` are dev deps.

---

Ah wait yes sorry lol